openpype/hosts/fusion/plugins/publish/collect_instances.py

```diff
@@ -22,25 +20,55 @@ def process(self, instance):
 
         # Include creator attributes directly as instance data
         creator_attributes = instance.data["creator_attributes"]
+        frame_range_source = creator_attributes.get("frame_range_source")
         instance.data.update(creator_attributes)
 
-        # Include start and end render frame in label
-        subset = instance.data["subset"]
+        # get asset frame ranges
         start = context.data["frameStart"]
         end = context.data["frameEnd"]
+        handle_start = context.data["handleStart"]
+        handle_end = context.data["handleEnd"]
+        start_handle = start - handle_start
+        end_handle = end + handle_end
+
+        if frame_range_source == "viewer_render_range":
+            # set comp render frame ranges
+            start = context.data["renderFrameStart"]
+            end = context.data["renderFrameEnd"]
+            handle_start = 0
+            handle_end = 0
+            start_handle = start
+            end_handle = end
+
+        if frame_range_source == "comp_range":
+            comp_start = context.data["compFrameStart"]
+            comp_end = context.data["compFrameEnd"]
+            render_start = context.data["renderFrameStart"]
+            render_end = context.data["renderFrameEnd"]
+            # set comp frame ranges
+            start = render_start
+            end = render_end
+            handle_start = render_start - comp_start
+            handle_end = comp_end - render_end
+            start_handle = comp_start
+            end_handle = comp_end
```

Comment:
> ALSO, be aware that the asset range might technically be incorrect for a subset if you're publishing **towards** another asset than the current context. E.g. from a comp you could be publishing towards different assets. When using the "asset's frame range" it might be unclear which asset frame range is actually being used for each instance?
> 
> Because it is **NOT** _from the asset database_ of the instance's asset but from the current context you're working in. This ultimately could get quite confusing. No?

Are you using it this way?

Reply:
We rarely publish to another asset and haven't done so in comp yet but have done so many times from Maya. So I'm not too bothered by this issue, but it's definitely a downside to this implementation.

But the question is which of the two you'd expect:
1. The asset range of the current working context
2. The asset range of the instance's asset.

Reply:
I just worked on a comp where I generated a sky image in a comp and needed it in multiple other comps. What I did was create a whole new asset for it but if I didn't have the power to do so, being able to publish to assets would be great!

Reply:
Ok I will rename the option to `Current asset context`.